### PR TITLE
Feature/make elastic pool optional

### PR DESCRIPTION
**Pull request by smneal**
- Adds the ability to specify whether a database is inside or outside the pool. It has been tested with standard tier SKU's.
- Also fixed the API version of the Redis deployment in the shared template as it was not valid.

The databaseNames variable needs the sku adding. So the current production variable will change from:

```
[{
	"name": "$(AuditDBName)",
	"maxSizeBytes": "268435456000"
}, {
	"name": "$(DevicesDBName)",
	"maxSizeBytes": "5368709120"
}, {
	"name": "$(DirectoriesDBName)",
	"maxSizeBytes": "5368709120"
}, {
	"name": "$(JobsDBName)",
	"maxSizeBytes": "5368709120"
}, {
	"name": "$(OrganisationsDBName)",
	"maxSizeBytes": "5368709120"
}]
```

to:

```
[{
	"name": "$(AuditDBName)",
	"maxSizeBytes": "268435456000",
	"sku": {
		"name": "S12",
		"tier": "Standard"
	}
}, {
	"name": "$(DevicesDBName)",
	"maxSizeBytes": "5368709120",
	"sku": {
		"name": "ElasticPool",
		"tier": "Standard"
	}
}, {
	"name": "$(DirectoriesDBName)",
	"maxSizeBytes": "5368709120",
	"sku": {
		"name": "ElasticPool",
		"tier": "Standard"
	}
}, {
	"name": "$(JobsDBName)",
	"maxSizeBytes": "5368709120",
	"sku": {
		"name": "ElasticPool",
		"tier": "Standard"
	}
}, {
	"name": "$(OrganisationsDBName)",
	"maxSizeBytes": "5368709120",
	"sku": {
		"name": "S12",
		"tier": "Standard"
	}
}]
```

### CODE REVIEW

File: Shared/shared.template.json [hunk 1]
```diff
@@ -348,7 +348,7 @@
       }
     },
     {
-      "apiVersion": "2018-03-01",
+      "apiVersion": "2017-05-10",
```
Comment (kalimakbar) [inline Shared/shared.template.json:351]: Why are we going back version wise?

Comment (OutKa5t) [inline Shared/shared.template.json:351]: This has already been changed. See https://github.com/DFE-Digital/login.dfe.infrastructure/commit/fe62a4f25e0fe91c4e1137b68e1875337f1efcbf

Comment (smneal) [reply Shared/shared.template.json]: I've updated that in https://github.com/DFE-Digital/login.dfe.infrastructure/pull/71/commits/9e08b25895567458df94af5a1ad9f61085ad6b4f. The change in master needs merging in to develop.

**Comment by smneal** [comment]
@DrogoNevets This also needs merging in to master as well

**Pull request merged**